Int flag update for control flow issue pyvisa/pyvisa-py#316

mwcmwc12 opened the pull request:
Per suggest in pyvisa/pyvisa-py#316, updating ControlFlow enum from
IntEnum to IntFlag.

<!--

Thanks for wanting to contribute to PyVISA :)

Here's some guidelines to help the review process go smoothly.

1. Please write a description in this text box of the changes that are being
   made.

2. Please ensure that the code is properly formatted and typed by running
   black, isort, flake8 and mypy. You can also use pre-commit hooks (see the
   developer documentation for detailed instructions)

3. Please ensure that you have written units tests for the changes made/features
   added.

4. If you are closing an issue please use one of the automatic closing words as
   noted here: https://help.github.com/articles/closing-issues-using-keywords/

5. Once review has taken place please do not add features or make changes out of
   the scope of those requested by the reviewer (doing this just add delays as
   already reviewed code ends up having to be re-reviewed/it is hard to tell
   what is new etc!).

Many thanks in advance for your cooperation!

-->

- [x] Closes pyvisa/pyvisa-py#316
- [x] Executed ``black . && isort -c . && flake8`` with no errors
- [ ] The change is fully covered by automated unit tests
- [ ] Documented in docs/ as appropriate
- [x] Added an entry to the CHANGES file

mwcmwc12 commented:
Ran testing against actual hardware on a Gwinstek PST3202. All expected results.

MatthieuDartiailh commented:
Can you add a changelog entry ?

mwcmwc12 commented:
Sorry, been super busy with work so this fell to the bottom of the priority list. I will try to update the changelog asap.

MatthieuDartiailh commented:
Sure, and can you do the same for https://github.com/pyvisa/pyvisa-py/pull/317 ?

mwcmwc12 commented:
Changelog updated for both PRs.

mwcmwc12 commented:
Fixed the mypy error finding by adding a new FlagAttribute class to attributes.py and changed AttrVI_ATTR_ASRL_FLOW_CNTRL to inherit from FlagAttribute.

Apologies for the problem. I need to speak with my co-worker whom I ask to help me test out the original IntFlag change why this problem didn't come up when testing with the actual hardware.

MatthieuDartiailh commented:
Don't worry about it. It would have had no runtime consequences hence your tests are not in cause. 

I will try to improve a bit the annotations to avoid the duplication but thanks for the patch.

MatthieuDartiailh commented:
Reducing the duplication would have some readability cost so I will just merge as is if the tests all pass.
